File: problem-10/solution.py
#!/usr/bin/env python


def primes(limit):
    """
    Return a generator for prime numbers.

    Args:
      limit (int): The maximun prime number

    This algorith uses the Sieve of Eratosthenes
    https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes
    """
    a = [True] * limit

    a[0] = False
    a[1] = False

    for i, is_prime in enumerate(a):
        if is_prime:
            yield i

            for n in range(i*i, limit, i):
                a[n] = False


if __name__ == '__main__':
    print(sum([p for p in primes(2000000)]))

File: problem-10/solution.lua
#!/usr/bin/lua

function iter_primes(limit)
    local primes = {false}
    local current = 1

    for i=2, limit do
        primes[i] = true
    end

    return function()
        for i=current, limit do
            if primes[i] then
                for j=i*i, limit, i do primes[j] = false end

                primes[i] = false
                current = i
                return i
            end
        end
    end
end

sum = 0
for prime in iter_primes(2000000) do
    sum = sum + prime
end

print(sum)
